tests/test_responses.py

```diff
@@ -307,6 +313,35 @@ async def app(scope, receive, send):
     client = test_client_factory(app)
     response = client.get("/")
     assert response.text == "Hello, world!"
+    assert (
+        response.headers["set-cookie"]
+        == "mycookie=myvalue; Domain=localhost; expires=Fri, 22 Jan 2100 12:00:10 GMT; "
+        "HttpOnly; Max-Age=10; Path=/; SameSite=none; Secure"
+    )
+
+
+@pytest.mark.parametrize(
+    "expires_factory",
+    [
+        lambda: datetime.now(timezone.utc) + timedelta(seconds=10),
+        lambda: format_datetime(
+            datetime.now(timezone.utc) + timedelta(seconds=10), usegmt=True
+        ),
+        lambda: 10,
+    ],
+)
+def test_expires_on_set_cookie(test_client_factory, expires_factory, time_machine):
+    time_machine.move_to(datetime(2100, 1, 22, 12, 0, 0))
+
+    async def app(scope, receive, send):
+        response = Response("Hello, world!", media_type="text/plain")
+        response.set_cookie("mycookie", "myvalue", expires=expires_factory())
+        await response(scope, receive, send)
+
+    client = test_client_factory(app)
+    response = client.get("/")
+    cookie: SimpleCookie = SimpleCookie(response.headers.get("set-cookie"))
+    assert cookie["mycookie"]["expires"] == "Fri, 22 Jan 2100 12:00:10 GMT"
```

Comment:
If someone can do better here, please feel free to suggest. @encode/maintainers

Reply:
I don't think we need the `timemachine` trick. Can we pass a well-defined datetime as input, rather than `datetime.now()`?

Reply:
Here I meant:

```python
@pytest.mark.parametrize(
    "expires, expires_output",
    [
        pytest.param(datetime.datetime(2023, 1, 1, 0, 0, 10, tzinfo=timezone.utc), "Fri, 22 Jan 2100 12:00:10 GMT", id="datetime"),
        pytest.param("Fri, 22 Jan 2100 12:00:10 GMT", "Fri, 22 Jan 2100 12:00:10 GMT", id="str"),
    ],
)

...

response.set_cookie("mycookie", "myvalue", expires=expires)

...

assert cookie["mycookie"]["expires"] == expires_output
```

(Looks like we ce can drop the `expires=10` case since it's tested above.)

Reply:
The goal of the assertion on the previous test is to make sure the header is the correct, and here is to check the expires field itself.

If I don't check with an integer, then I don't need the time-machine here, but I think I still need it for the above... 🤔